src/wallet/redux/reducers/__tests__/challenging.test.ts

```diff
@@ -0,0 +1,127 @@
+import { walletReducer } from '..';
+
+import * as states from '../../../states';
+import * as actions from '../../actions';
+
+import { itTransitionsToStateType, itSendsATransaction } from './helpers';
+import { scenarios } from '../../../../core';
+import { ChallengeProof } from 'src/wallet/domain/ChallengeProof';
+const {
+    asPrivateKey,
+    revealHex,
+    acceptHex,
+    acceptSig,
+    proposeHex,
+    proposeSig,
+    participants,
+    channelId,
+    channelNonce,
+    libraryAddress,
+} = scenarios.standard;
+
+const defaults = {
+    uid: 'uid',
+    participants,
+    libraryAddress,
+    channelId,
+    channelNonce,
+    lastPosition: revealHex,
+    penultimatePosition: acceptHex,
+    turnNum: 6,
+    adjudicator: 'adj-address',
+    ourIndex: 0,
+    address: 'address',
+    privateKey: asPrivateKey,
+    challengeProof: new ChallengeProof(proposeHex, acceptHex, proposeSig, acceptSig),
+
+};
+
+
+describe('when in APPROVE_CHALLENGE', () => {
+    const state = states.approveChallenge({ ...defaults });
+    describe('when a challenge is approved', () => {
+        const action = actions.approveChallenge();
+        const updatedState = walletReducer(state, action);
+        itTransitionsToStateType(states.INITIATE_CHALLENGE, updatedState);
+        itSendsATransaction(updatedState);
```

Comment:
@lalexgap Your browser seems to still default to 4 spaces instead of two .. :(

Reply:
Hmm, do you only get 2 when you run auto-format for describe blocks?

Reply:
I have `"editor.tabSize": 2,` in my user settings.

Reply:
How do I run auto-format?

Reply:
Looks like my tabSize was set to 4. Had to set to 2, turn off auto-detect, and then run auto-format.